app/controllers/documents_controller.rb

```diff
@@ -101,7 +101,16 @@ def document_error_messages
   end
 
   def fetch_document
-    @document = document_klass.find(params[:content_id])
+    begin
+      @document = document_klass.find(params[:content_id])
+    rescue Document::RecordNotFound => e
+      flash[:danger] = "Document not found"
+      redirect_to documents_path(document_type: document_type)
+
+      Airbrake.notify(e)
+
+      false
```

Comment:
You don't need to return false in filters any more to halt the chain; a call to `render`, `redirect_to` or `head` will do that for you.